packages/proxy/lib/http/response-middleware.ts

```diff
@@ -381,11 +382,73 @@ const MaybePreventCaching: ResponseMiddleware = function () {
   this.next()
 }
 
+const determineIfNeedsMultiDomainHandling = (ctx) => {
+  const previousAUTRequestUrl = ctx.getPreviousAUTRequestUrl()
+
+  // A cookie needs multi-domain handling if it's an AUT request and
+  // either the request itself is cross-origin or the origins between
+  // requests don't match, since the browser won't set them in that
+  // case and if it's secondary-domain -> primary-domain, we don't
+  // recognize the request as cross-origin
+  return (
+    !!ctx.req.isAUTFrame &&
+    (
+      (previousAUTRequestUrl && !cors.urlOriginsMatch(previousAUTRequestUrl, ctx.req.proxiedUrl))
+      || !reqMatchesOriginPolicy(ctx.req, ctx.getRemoteState())
+    )
+  )
+}
+
+const cookieSameSiteRegex = /SameSite=(\w+)/i
+const cookieSecureRegex = /Secure/i
+const cookieSecureSemicolonRegex = /;\s*Secure/i
+
+const ensureSameSiteNone = ({ cookie, browser, isLocalhost, url }) => {
+  debug('original cookie: %s', cookie)
+
+  if (cookieSameSiteRegex.test(cookie)) {
+    debug('change cookie to SameSite=None')
+    cookie = cookie.replace(cookieSameSiteRegex, 'SameSite=None')
+  } else {
+    debug('add SameSite=None to cookie')
+    cookie += '; SameSite=None'
+  }
+
+  const isFirefox = browser.family === 'firefox'
+
+  // Secure is required in Chrome for SameSite=None cookies to be set, but if
+  // in Firefox and http://localhost, it causes the cookie not to be set,
+  // so we need to ensure there is no Secure in that case
```

Comment:
i'd also like to see some links to defintiive documentation (RFC, source code, authoritative reference...) for this, i believe you, but it's important to clearly track browser errata like this since it changes under our feet all the time...

Reply:
I dug a bit deeper into this and the reason there's a difference in Firefox is that it doesn't consider `http://localhost` to be a [secure context](https://w3c.github.io/webappsec-secure-contexts/#is-origin-trustworthy) when loaded by the Cypress-launched browser, even though it's [normally considered a secure context](https://w3c.github.io/webappsec-secure-contexts/#localhost). 

Here's an [existing issue about it](https://github.com/cypress-io/cypress/issues/18217). It has something to do with the way we launch Firefox. I dug into it a little bit, but it needs a bit more research and I decided to time-box the effort since it's out of scope for this work. 

The fact is that `http://localhost` won't allow `SameSite=None; Secure` cookies to be set even outside of multi-domain. The question then is whether we should have this workaround at all to support it for multi-domain when it doesn't even work in general. I decided to keep the workaround in for the following reasons:

- It's only going in the experimental release and will be removed once [better cookie handling](https://github.com/cypress-io/cypress/issues/20685) is implemented.
- The primary use-case for multi-domain is authentication, so I think it's important to get cookies working consistently between browsers for authentication purposes.
- We may need a similar workaround outside of multi-domain anyway, though I don't want to go that far at this moment. I feel it's best to scope this to the multi-domain work and experimental release instead of doing a general workaround without doing more research into why Firefox doesn't consider the browser a secure context.

I [improved the comment here](https://github.com/cypress-io/cypress/pull/20450/commits/aeed87ed8d40eda1a65a1e9b42b79904cc00110a), explaining what I did above in brief and linking to the spec regarding secure contexts.